calculator_test.go

```diff
@@ -2,23 +2,288 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	tests := []struct {
+		in1, in2, want float64
+		expectsErr     bool
+		name           string
+	}{
+		{in1: 0, in2: 0, want: 0, name: "Add two units"},
```

Comment:
It's a good idea to put the `name` first, which is the logical order we'd read it in, isn't it?